db-service/lib/cqn4sql.js

```diff
@@ -911,7 +911,7 @@ function cqn4sql(originalQuery, model = cds.context?.model || cds.model) {
       const { index, tableAlias } = inferred.$combinedElements[k][0]
       const element = tableAlias.elements[k]
       // ignore FK for odata csn / ignore blobs from wildcard expansion
-      if (isODataFlatForeignKey(element) || (element['@Core.MediaType'] && !element['@Core.IsURL'])) return
+      if (isManagedAssocInFlatMode(element) || (element['@Core.MediaType'] && !element['@Core.IsURL'])) return
```

Comment:
rather filter out duplicates than to rely on some meta information w.r.t. the model